src/components/Accordion/Accordion.tsx

```diff
@@ -138,30 +143,46 @@ const getElementsFromData = (
               <AccordionButton
                 aria-label={finalAriaLabel}
                 id={`${id}-button-${index}`}
-                borderColor={
-                  isDarkMode ? "dark.ui.border.default" : "ui.gray.medium"
-                }
                 padding={multiplePadding}
                 ref={content.buttonInteractionRef}
+                // Fix for double border issue in non-hovered state
+                // i.e. Hide the bottom border unless the accordion is last or expanded
+                borderBottomColor={
+                  isLast || isExpanded ? "ui.gray.medium" : "transparent"
+                }
+                // Fix for double border issue on hover
+                // i.e. Hide the top border on the next button after the hovered button unless it's first
+                borderTopColor={
+                  !(index === 0) &&
+                  index === hoveredButtonIndex + 1 &&
+                  "transparent"
```

Comment:
This works but make it a tertiary to follow the general convention. The false case would be undefined.